src/main/Main.ts

```diff
@@ -512,6 +511,59 @@ export function main(init: Init): void {
     });
   }
 
+  function setProtocolRegistrationState(registered: boolean) : boolean {
+    // Check how the Launcher was started
+    const procDefault = process.defaultApp;
+    const procArgCount = process.argv.length;
+
+    if (procDefault && procArgCount < 2) {
+      return true; // Don't need to change
+    }
+
+    /*
+     * The return value of app.removeAsDefaultProtocolClient() is really inconsistent
+     * between platforms, so we wrap it to consistently match:
+     *
+     * true - The app was set as the default handler for 'protocol' and was successfully
+     *        removed, or was not the default handler in the first place.
+     * false - The app was set as the default handler for 'protocol' and there was an
+     *         issue removing it.
+     */
+    function normalizedRemove(protocol: string, path: string, args: string[]) : boolean {
+      const needsRemove = app.isDefaultProtocolClient(protocol, path, args);
+      if (!needsRemove || process.platform === 'linux') {
+        /*
+         * Electron has not implemented app.removeAsDefaultProtocolClient() on Linux so
+         * it always fails; however, for our purposes we return true as we've done all
+         * we can and nothing unexpected has occurred.
+         *
+         * https://github.com/electron/electron/blob/a867503af63bcf24f935ae32fc8d88fe5e7a786a/shell/browser/browser_linux.cc#L118
+         */
+        return true;
+      }
+
+      return app.removeAsDefaultProtocolClient(protocol, path, args);
+    }
+
+    // Add/remove protocol registration
+    type ProtocolFunction = (protocol: string, path: string, args: string[]) => boolean;
+
+    const func: ProtocolFunction = registered ? app.setAsDefaultProtocolClient : normalizedRemove;
+    const verb = registered ? 'set' : 'unset';
+    const scheme = 'flashpoint';
+    const pPath = process.execPath;
+    const pArgs = procDefault ? [path.resolve(process.argv[1])] : [];
```

Comment:
Potentional out of bounds error

Reply:
`process.argv[1]` is never evaluated when that would be OOB since L555 isn't reached with `procDefault` as true and `argv`'s count less than 2 due to the conditional on L519.

If you want, I can repeat the count check as part of the ternary on L555 for posterity/style since that's not necessarily immediately obvious when skimming the function.